chapters/en/slides/module3/module3_01.Rmd

```diff
@@ -0,0 +1,274 @@
+by composing it with a mark and a single encoding,
+instead of remembering a specific name.
+
+---
+
+## Thinner marks also saturate for large data sets
+
+```{python}
+alt.Chart(movies).mark_tick().encode(
+    x='runtime')
+```
+
+Notes:
+However,
+when we increase the number of observations even further (to about 500)
+it becomes difficult to tell exactly how the data distribution looks,
+even with these thin marks.
+
+---
+
+## Using histograms to visualize distributions avoids saturation issues
+
+```{python}
+alt.Chart(movies).mark_bar().encode(
+    alt.X('runtime', bin=alt.Bin(maxbins=30)),
+    y='count()')
+```
+
+Notes:
+Instead of visualizing each individual observation,
+we can use the histograms we learnt about in last module,
+where we divide one axis into groups (or "bins")
+and represent the count of observations in each group with bar.
+
+In this plot,
+we can directly see that most movies are around 100 min in length.
+But from our experience,
+we know that factors such as genre and production year,
+can influence movie length.
+
+How can we separate the data based on other columns in the dataset
+and compare multiple distributions?
+
+---
+
+## Comparing multiple distributions via faceting
+
+```{python}
+(alt.Chart(movies).mark_bar().encode(
+    alt.X('runtime', bin=alt.Bin(maxbins=30)),
+    y='count()')
+ .facet('country'))
+```
+
+Notes:
+Let's start by comparing the movie length broken down per country.
+Here we use faceting which we learnt about in the last module,
+to create one histogram per country.
+
+You can see that this dataset only contains movies made in the UK or the US.
+It does seem like the distribution of movies is pretty equal between the two countries,
+but it is a bit hard to see because there are so many more movies made in the US.
+
+---
+
+## Comparing multiple distributions via faceting with independent y-axes
+
+```{python}
+(alt.Chart(movies).mark_bar().encode(
+    alt.X('runtime', bin=alt.Bin(maxbins=30)),
+    y='count()')
+ .facet('country')
+ .resolve_scale(y='independent'))
+```
+
+Notes:
+By default when faceting,
+the y-axis is the same for all the subplots.
+This is often the most suitable choice,
+because it gives us an idea of both the shape of the distribution
+and the total number of observations.
+
+If we want to drill down and compare the distribution shape more closely
+regardless of the number of observations
+we could define the y-axes to be independent for each subplot,
+instead of linked to the same axis range.
+
+Now we can see that the distributions might be a bit different,
+it looks like the UK one is slightly bimodal (it has two peaks),
+whereas the US one is unimodal (it has one peak).
+
+We could explore that data further to find out why this is,
+e.g. maybe the genre is more related to movie length
+and the UK movies in this dataset are of different genres than the US ones?
+
+---
+
+## Comparing multiple distributions grouped by two variables
```

Comment:
Is there another country in the data set you could compare? What about China or India? They produce a lot of movies? Which might show more differences to the US movies? Otherwise, I think we need to temper our conclusions on how different these distributions are, as again, I would not support the claim that these differences are meaningful.

Reply:
I rephrased to say that we can't claim that these differences are meaningful without more data and more plots with different bin sizes. There is unfortunately even fewer movies from other countries in the original dataset and I think none from the two you mentioned. It would be interesting, but I don't think we have time to find that data now.